### ISSUE TITLE
[SideNav] Add a focus class when sidenav item has focus

### ISSUE THREAD

**Issue by davidicus** [IBM/carbon-components-react#2148]
### Summary

Focus class is needed to expand the sidenav when sidenav items contain focus. Should be similar to hover state for keyboard users.  Currently you are able to navigate through keyboard and even open subnavs. This causes a bug where the focused item is not visible. Additionally the SideNavMenu collapses and hides its icon when its child list is open. 

### Justification

For accessibility this should open when items are focused. especially hidden subnavs

### Available extra resources

I have opened a PR for this work.

### Extra notes
These full completion of this feature would need changes to the carbon component and carbon styles.
